toplevel/genprintval.ml

```diff
@@ -511,7 +511,7 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
               in
               (lid, v) :: tree_of_fields (pos + 1) remainder
         in
-        Oval_record (tree_of_fields pos lbl_list)
```

Comment:
tree_of_fields could be rewritten with mapi. Also should oval_record use non empty list